scripts/Interface/ui/drill/model/DrillSample.py

```diff
@@ -5,87 +5,349 @@
+
+    """
+    Name of the group, if the sample is in a group.
+    """
+    _group = None
+
+    """
+    Index of the sample in its group.
+    """
+    _groupIndex = None
+
+    """
+    True if the sample is the master sample of its group.
+    """
+    _master = None
+
+    _status = None
+
+    """
+    Controller for the parameters.
+    """
+    _controller = None
+
+    """
+    Sent if the group changed.
+    """
+    groupChanged = Signal()
+
+    """
+    Sent when a new parameter is added.
+    Args:
+        DrillParameter: the new parameter
+    """
+    newParameter = Signal(DrillParameter)
+
+    """
+    Signals that the status of the sample changed.
+    """
+    statusChanged = Signal()
+
+    def __init__(self, index):
         """
         Create an empty sample.
         """
+        super().__init__()
         self._parameters = dict()
+        self._index = index
+        self._group = None
+        self._master = False
+
+    def setController(self, controller):
+        """
+        Controller for the parameters.
+
+        Args:
+            controller (DrillParameterController): parameter controller
+        """
+        self._controller = controller
+
+    def setIndex(self, index):
+        """
+        Set the sample index.
+
+        Args:
+            index (int): index of the sample
+        """
+        self._index = index
+
+    def getIndex(self):
+        """
+        Get the sample index.
 
-    def setParameters(self, parameters):
+        Returns:
+            int: index of the sample
+        """
+        return self._index
+
+    def setName(self, name):
         """
-        Set the processing parameters.
+        Set the sample name.
 
         Args:
-            parameters (dict(str:str)): parameter key:value pairs
+            name (str): name of the sample
         """
-        self._parameters = {k:v for k,v in parameters.items()}
+        self._name = name
 
-    def getParameters(self):
+    def getName(self):
         """
-        Get the processing parameters.
+        Get the sample name.
 
         Returns:
-            dict(str:str): parameter key:value pairs
+            str: name of the sample
         """
-        return {k:v for k,v in self._parameters.items()}
+        return self._name
+
+    def setOutputName(self, name):
+        """
+        Set the name of the output workspace.
+
+        Args:
+            name (str): name
+        """
+        self._outputName = name
+
+    def getOutputName(self):
+        """
+        Get the name of the output workspace.
+
+        Returns:
+            str: name
+        """
+        return self._outputName
+
+    def setGroup(self, group, index=None):
+        """
+        Set the group and index of the sample.
+
+        Args:
+            group (str): name of the group. None if the sample is not in a
+                         group.
+            index (int): index of the sample in its group
+        """
+        if not group:
```

Comment:
If both if/else branches are non-empty, it is more readable to have the direct condition. Double negation causes a delay in reading the code.